### PR TITLE
INTLY-9315 - Refine critical SLO summary dashboard and fix non-instant alert state updates

### PR DESCRIPTION
**Pull request by HubertStefanski**
# Description
- Adding 28 day limit on the queries to adhere to the SLA, especially in %SLO panel, previously the SLO percentage was non-indicative of the actual state, potentially miscalculating the SLA breach.
- Add instant selector to fetch alert states as they change, rather than previous approach of waiting 15 minutes.
- Add default dashboard refresh to 10 seconds, previously no refresh period was set and the dashboard would only update on manual refresh.
- Change default time range to now-5m (previously now-15m) meaning that states only got updated after 15 minutes from the actual change, no longer an issue due to `instant : true`, (now-5m) is still needed to update line graphs, the instant field on that query will break the graph.

## JIRA
https://issues.redhat.com/browse/INTLY-9315

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

## Verification
- Install this branch on your cluster.
- Run [E04](https://issues.redhat.com/browse/INTLY-9343) and ensure that the dashboard reflects the state of the alerts as seen in Prometheus UI
- Alerts should be updated immediately
- Critical SLO summary graph and % should grab metrics from the last 28 days
- Use the admin secret on the cluster to get an admin view of the graphs

### CODE REVIEW

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go [hunk 1]
```diff
@@ -225,6 +226,7 @@ const MonitoringGrafanaDBCriticalSLOAlertsJSON = `{
 			"targets": [{
 				"expr": "sum(ALERTS{severity='critical', alertstate='firing'}) or vector(0)",
 				"format": "time_series",
+				"instant": false,
```
Comment (HubertStefanski) [inline pkg/products/monitoring/dashboards/criticalSLOAlerts.go:229]: Added in these instant fields even for places where they're not being used, I think it adds a bit more readability to the already busy Json, but I can take them out if we don't want them here.

Comment (HubertStefanski) [inline pkg/products/monitoring/dashboards/criticalSLOAlerts.go:229]: Added in these instant fields even for places where they're not being used, I think it adds a bit more readability to the already busy Json, but I can take them out if we don't want them.

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go [hunk 2]
```diff
@@ -4181,8 +4212,9 @@ const MonitoringGrafanaDBCriticalSLOAlertsJSON = `{
 			}
 		]
 	},
+	"refresh": "10s",
```
Comment (HubertStefanski) [inline pkg/products/monitoring/dashboards/criticalSLOAlerts.go:4215]: Aforementioned default refresh, previously the dashboard wouldn't update automatically, now it does so every 10 seconds, looking for opinions on this time, should it be lower/higher?

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go [hunk 3]
```diff
@@ -4181,8 +4212,9 @@ const MonitoringGrafanaDBCriticalSLOAlertsJSON = `{
...
 	"time": {
-		"from": "now-15m",
+		"from": "now-5m",
```
Comment (HubertStefanski) [inline pkg/products/monitoring/dashboards/criticalSLOAlerts.go:4217]: From what I can tell, this is the default "poll rate" that the graphs will use to update themselves if no `instant` field is set, I might be wrong, just going off my observations.

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 1 above)
Comment (obrienrobert) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:229]: Is this default value of `instant` false? If so, I'm just thinking if there is a need to add this to each `target` block. I would be thinking along the lines of why are we explicitly adding this variable with it's default value but not other doing the same for other possible values?

Comment (obrienrobert) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:229]: Is the default value of `instant` false? Just thinking if we need to set it is so, as we would be explicitly setting the default value for a single variable, but not for other possible variables.

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 2 above)
Comment (obrienrobert) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:4215]: Was not having a `refresh` value set causing issues/problems? Or is this just more of an enhancement?

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 1 above)
Comment (HubertStefanski) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:229]: Yes, the default value for instant is false, I get your point, in this case it's probably best to remove them if we're not using those, thanks for the feedback :+1:

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 2 above)
Comment (HubertStefanski) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:4215]: It wasn't causing problems with the metrics themselves right away, it just meant that the dashboard wouldn't be updated unless the refresh button was clicked, so this is a 50/50 bug fix/enhancement, most of the other dashboards have a default automatic refresh, so I think we're probably best off having an automatic refresh for this one as well, seeing as this is a what I would consider "vital" dashboard we should be getting an update quite often. This is also something that people might not be aware of when testing E04 during RC testing, I usually refreshed mine manually and wasn't aware that there was an option to set an automatic refresh until I looked into the JSON.

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 3 above)
Comment (obrienrobert) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:4217]: Looks like this sets the time range value? https://grafana.com/docs/grafana/latest/dashboards/time-range-controls/#common-time-range-controls

from

I'm not sure what is the ideal value is for this, if this is the case.

**Comment by obrienrobert** [comment]
@HubertStefanski I've noticed on some of the panels that the `Last 28 days` text overlaps the panel title. I've attached a screenshot. Not sure if this is new or if there was always text overlapping, but it's just something I observed.

slo_dashboard_alerts

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 2 above)
Comment (obrienrobert) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:4215]: Agree, sounds reasonable. At the minimum it reduces the need to refresh to update the info.

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 3 above)
Comment (HubertStefanski) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go:4217]: So I think what this does is that it checks if the alert was firing in any period from now to 5 minutes ago then uses that to determine whether to update dashboards which don't use instant fields, not so much of a problem now though I think with the auto-refresh

**Comment by HubertStefanski** [comment]
> @HubertStefanski I've noticed on some of the panels that the `Last 28 days` text overlaps the panel title. I've attached a screenshot. Not sure if this is new or if there was always text overlapping, but it's just something I observed.

That's a result of setting the `timefrom` field in the dashboard JSON, I'll have a look if I can move it somehow, or at least make the box big enough so that it doesn't obstruct.

**Comment by HubertStefanski** [comment]
image
fixed the issue @obrienrobert pointed out above

**Pull request merged**